docs/aws_iam_role_crd.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: awsiamroles.amazonaws.com
+spec:
+  group: amazonaws.com
+  version: v1
+  scope: Namespaced
+  names:
+    kind: AWSIAMRole
+    singular: awsiamrole
+    plural: awsiamroles
+    categories:
+    - all
+  additionalPrinterColumns:
+  - name: RoleARN
+    type: string
+    description: Full RoleARN
+    JSONPath: .status.roleARN
+  - name: Expiration
+    type: string
+    description: Expiration time of the current credentials provisioned for the role
+    JSONPath: .status.expiration
+  subresources:
+    # status enables the status subresource.
+    status: {}
+  # validation depends on Kubernetes >= v1.11.0
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            roleReference:
+              description: |
+                Reference to an AWS IAM role which can either be a role name
+                or a full IAM role ARN.
+              type: string
+              minLength: 3
+            roleSessionDuration:
+              description: |
+                Specify the tole session duration in seconds. defaults to 3600
+                seconds (1 hour). This value must be less than or equal to the
+                `MaxSessionDuration` value of the IAM role.
+              type: integer
+              minimum: 900   # 15 minutes
+              maximum: 43200 # 12 hours
+            roleDefinition:
```

Comment:
Is this CRD defining a new IAM role, or is this a snapshot of an existing IAM role as read from AWS?

Reply:
Yes, the idea, which is not yet implemented and won't be in this PR, is to allow users to define an IAM role via the CRD as an alternative to referencing it. This would then take care of provisioning the IAM role with the correct assume role configuration etc.

I will remove this from this PR, was just easier to design the whole thing with this idea in mind.